src/start_root.c

```diff
@@ -37,8 +38,15 @@ void __sel4_start_root(seL4_BootInfo *boot_info)
 {
     sel4runtime_uintptr_t tdata_start = (sel4runtime_uintptr_t) &_tdata_start[0];
     sel4runtime_uintptr_t tdata_end = (sel4runtime_uintptr_t) &_tdata_end[0];
+    sel4runtime_uintptr_t tbss_start = (sel4runtime_uintptr_t) &_tbss_start[0];
     sel4runtime_uintptr_t tbss_end = (sel4runtime_uintptr_t) &_tbss_end[0];
 
+    /* The .tdata section is empty and hence, the linker script variables are zero */
+    if (tdata_start == 0) {
+      tdata_start = tbss_start;
+      tdata_end = tbss_start;
+    }
```

Comment:
Are you able to expand on this comment describing how this is added initially to handle a crash under lld where the variables are 0 when the sections are empty?

Reply:
Thanks!